nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/processors/ConsumeJMS.java

```diff
@@ -375,4 +507,11 @@ private FlowFile updateFlowFileAttributesWithJMSAttributes(Map<String, String> j
         flowFile = processSession.putAllAttributes(flowFile, attributes);
         return flowFile;
     }
+
+    private Map<String, String> mergeJmsAttributes(Map<String, String> headers, Map<String, String> properties) {
+        final Map<String, String> jmsAttributes = new HashMap<>(headers);
```

Comment:
Good catch. The issue is that it is also a problem in the case of standard processing. I will log a warning in that case too, but the correct solution would be to prefix those properties. Unfortunately, I cannot do it now because of backward compatibility reasons, so I submitted a new ticket targeting only 2.0 to fix that! https://issues.apache.org/jira/browse/NIFI-11350